lib/streamlit/runtime/app_session.py

```diff
@@ -883,8 +883,6 @@ def _populate_config_msg(msg: Config) -> None:
     msg.max_cached_message_age = config.get_option("global.maxCachedMessageAge")
     msg.allow_run_on_save = config.get_option("server.allowRunOnSave")
     msg.hide_top_bar = config.get_option("ui.hideTopBar")
-    # ui.hideSidebarNav is deprecated, will be removed in the future
-    msg.hide_sidebar_nav = config.get_option("ui.hideSidebarNav")
```

Comment:
nit: the comment in the proto file probably also needs to be updated to `client.showSidebarNavigation`:

https://github.com/streamlit/streamlit/blob/173bfb82f6fdf95107b39716dc231a6ca0ba3063/proto/streamlit/proto/NewSession.proto#L116